nifi-nar-bundles/nifi-workday-bundle/nifi-workday-processors/src/main/java/org/apache/nifi/processors/workday/GetWorkdayReport.java

```diff
@@ -0,0 +1,432 @@
+        RecordSetWriterFactory recordSetWriterFactory = context.getProperty(RECORD_WRITER_FACTORY).asControllerService(RecordSetWriterFactory.class);
+        WebClientService webClientService = standardWebClientServiceProvider.getWebClientService();
+        webClientAtomicReference.set(webClientService);
+        recordReaderFactoryAtomicReference.set(recordReaderFactory);
+        recordSetWriterFactoryAtomicReference.set(recordSetWriterFactory);
+    }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        FlowFile flowfile = session.get();
+
+        if (skipExecution(context, flowfile)) {
+            return;
+        }
+
+        ComponentLog logger = getLogger();
+        FlowFile responseFlowFile = null;
+
+        try {
+            WebClientService webClientService = webClientAtomicReference.get();
+            URI uri = new URI(context.getProperty(REPORT_URL).evaluateAttributeExpressions(flowfile).getValue().trim());
+            long startNanos = System.nanoTime();
+            String authorization = createAuthorizationHeader(context, flowfile);
+
+            try(HttpResponseEntity httpResponseEntity = webClientService.get().uri(uri).header(HEADER_AUTHORIZATION, authorization).retrieve()) {
+                responseFlowFile = createResponseFlowFile(flowfile, session, context, httpResponseEntity);
+                long elapsedTime = TimeUnit.NANOSECONDS.toMillis(System.nanoTime() - startNanos);
+                Map<String, String> commonAttributes = createCommonAttributes(uri, httpResponseEntity, elapsedTime);
+
+                if (flowfile != null) {
+                    flowfile = session.putAllAttributes(flowfile, decorateWithMimeAttribute(commonAttributes, httpResponseEntity));
+                }
+                if (responseFlowFile != null) {
+                    responseFlowFile = session.putAllAttributes(responseFlowFile, commonAttributes);
+                    if (flowfile != null) {
+                        session.getProvenanceReporter().fetch(responseFlowFile, uri.toString(), elapsedTime);
+                    } else {
+                        session.getProvenanceReporter().receive(responseFlowFile, uri.toString(), elapsedTime);
+                    }
+                }
+
+                route(flowfile, responseFlowFile, session, context, httpResponseEntity.statusCode());
+            }
+        } catch (Exception e) {
+            if (flowfile == null) {
+                logger.error("Request Processing failed", e);
+                context.yield();
+            } else {
+                logger.error("Request Processing failed: {}", flowfile, e);
+                session.penalize(flowfile);
+                flowfile = session.putAttribute(flowfile, GET_WORKDAY_REPORT_JAVA_EXCEPTION_CLASS, e.getClass().getName());
```

Comment:
Recommend using `getClass().getSimpleName()`. Although this is less specific, it avoids tighter coupling to the specific Exception class.